ndb/src/google/cloud/ndb/_eventloop.py

```diff
@@ -259,6 +261,28 @@ def run(self):
                 break
 
 
+class _LocalContexts(threading.local):
+    """Maintain a thread local stack of event loops."""
+
+    def __init__(self):
+        self.stack = []
+
+    def push(self):
+        loop = EventLoop()
+        self.stack.append(loop)
+        return loop
+
+    def pop(self):
+        return self.stack.pop(-1)
```

Comment:
It is a little weird to have the `push` / `pop` names, without their "normal" stack semantics (user providing a widget in `push`, versus creating one).  Could we have the context manager instantiate an `EventLoop` and pass it in to `push`?

Reply:
True. I had it that way at first and then changed for reasons that are obscure to me, even. I've changed it back.